UI/MapWnd.cpp

```diff
@@ -322,7 +325,38 @@ class MapWnd::MapScaleLine : public GG::Control {
         glLineWidth(1.0);
     }
 
-    void Update(double zoom_factor) {
+    GG::X GetLength() {
+        return m_line_length;
+    }
+
+    void Update(double zoom_factor, std::set<int>& fleet_ids, int sel_system_id) {
```

Comment:
Could you add some comments before and within this code to explain what it's doing?